blog-post/blog-post.md

```diff
@@ -0,0 +1,522 @@
+
+### Compile Time Evaluation
+
+While not yet implemented, we are planning to add a robust system for compile time evaluation. We
+expect to support at least the compile time evaluation of pure expressions that do not require
+access to memory. Some key design questions that remain open relate to the degree to which we will
+support access to memory at compile time, the need for compile time specific primitives (e.g. string
+concatenation / hashing), as well as the need for reflection.
+
+We want to make sure the needs of the community are met here. Please feel free to reach out to use
+on Github / The Solidity Forum, or our Matrix channel if you have concrete use cases in mind for
+this feature.
+
+We will publish more once we have a concrete design / prototype implementation.
+
+## Current status
+
+Core Solidity prototype is being actively developed by Argot's Programming Languages
+Research team and it currently supports the following features:
+
+- Definition of algebraic data types, pattern matching, type classes and polymorphic functions.
+
+- Data Locations represented as types: Data locations (e.g., storage, memory)
+  are now just standard library types. This enables the creation of composite
+  types with mixed locations, such as `Broker` type which holds references to storage
+  arrays and memory data, as presented in the next code piece:
+
+```
+data Broker = Broker (memory(word), storage(array(address, word)))
+```
+
+- Classic Solidity compatible ABI encoding / decoding, mappings and contract constructors.
+
+Using these features, we can implement a simple [ERC20 contract in Core Solidity](erc20.sol).
+
+## Compatibility and Interoperability
+
+Introducing such a major revision to any programming language is challenging. While a certain degree
+of breakage is inevitable (and even desired), we want to make the transition as smooth as possible
+and avoid a split in the language.
+
+We intend to keep the syntax breakage as small as possible, and expect to make significant
+alterations to the current prototype to bring syntax much more in line with Classic Solidity.
+
+As with previous breaking upgrades to Solidity, ABI compatibility will be maintained between
+versions, allowing individual contracts written in incompatible versions to interoperate and live
+side by side in the same project (this strategy is also used by Rust with their "Editions" feature).
+Thanks to our history of releasing breaking changes, much of the ecosystem tooling already supports
+multiple incompatible language versions in the same project (TODO: check and confirm the state of
+things in foundry / hardhat).
+
+We are also investigating the feasibility of deeper interoperability beyond just the contract ABI.
+We expect that it will be possible to share at least free functions and interface definitions
+between language versions. We will write more concretely on this topic when we have detailed designs
+or working prototypes to share.
```

Comment:
I mean, we will also publish the blog post about the transition so we will share a bit more detail even before then. It won't be a detailed design of course, but still.